storage/src/mmr/verification.rs

```diff
@@ -1,25 +1,34 @@
 use crate::mmr::hasher::Hasher;
-use crate::mmr::iterator::PeakIterator;
+use crate::mmr::iterator::{PathIterator, PeakIterator};
 use bytes::{Buf, BufMut};
 use commonware_cryptography::{Array, Hasher as CHasher};
 use commonware_utils::SizedSerialize;
 
-/// A `Proof` contains the information necessary for proving the inclusion of an element, or some
-/// range of elements, in the MMR from its root hash. The `hashes` vector contains: (1) the peak
-/// hashes other than those belonging to trees containing some elements within the range being
-/// proven, followed by: (2) the nodes in the remaining perfect trees necessary for reconstructing
-/// their peak hashes from the elements within the range. Both segments are ordered by decreasing
-/// height.
+/// Contains the information necessary for proving the inclusion of an element, or some range of
+/// elements, in the MMR from its root hash.
+///
+/// The `hashes` vector contains: (1) the peak hashes other than those belonging to trees containing
+/// some elements within the range being proven, followed by: (2) the nodes in the remaining perfect
+/// trees necessary for reconstructing their peak hashes from the elements within the range. Both
+/// segments are ordered by decreasing height.
 #[derive(Clone, Debug, Eq)]
-/// A Proof contains the information necessary for proving the inclusion of an element, or some
-/// range of elements, in the MMR.
 pub struct Proof<H: CHasher> {
     /// The total number of nodes in the MMR.
     pub size: u64,
-    /// The hashes necessary for proving the inclusion of an element, or range of elements, in the MMR.
+    /// The hashes necessary for proving the inclusion of an element, or range of elements, in the
+    /// MMR.
     pub hashes: Vec<H::Digest>,
 }
 
+// A trait that allows generic generation of an MMR inclusion proof.
+pub trait Provable<H: CHasher> {
```

Comment:
If this is now just `size`, `get_node`, I think we should call it `Storage`.

Reply:
changed